assets/js/components/dashboard-sharing/ModuleRecoveryAlert/RecoverableActions.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * ModuleRecoveryAlert RecoverableActions component.
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { without } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment, useCallback, useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect, useDispatch } from 'googlesitekit-data';
+import { Checkbox } from 'googlesitekit-components';
+import { CORE_MODULES } from '../../../googlesitekit/modules/datastore/constants';
+import { DAY_IN_SECONDS } from '../../../util';
+import Errors from './Errors';
+import ActionsCTALinkDismiss from '../../../googlesitekit/notifications/components/common/ActionsCTALinkDismiss';
+
+export default function RecoverableActions( {
+	id,
+	recoverableModules,
+	userRecoverableModuleSlugs,
+	hasMultipleRecoverableModules,
+} ) {
+	const [ selectedModuleSlugs, setSelectedModuleSlugs ] = useState( null );
```

Comment:
I changed the `checked` object to a list of selected slugs. I think it makes the `onChange` much more clear as well as the function which handles the recover CTA. Everywhere else we were constantly converting it to a list of keys and filtering to those that were "checked" so it made a lot more sense to store it as this list instead.